lib/puppet/settings/file_setting.rb

```diff
@@ -135,7 +130,7 @@ def to_resource
     # Make sure the paths are fully qualified.
     path = File.expand_path(path)
 
-    return nil unless type == :directory or create_files? or Puppet::FileSystem.exist?(path)
+    return nil unless type == :directory or Puppet::FileSystem.exist?(path)
```

Comment:
Should we update this line and the one below to use `||` instead of `or`?

Reply:
Updated